src/sql/src/plan/statement/ddl.rs

```diff
@@ -289,6 +290,16 @@ pub fn plan_create_table(
     } else {
         scx.allocate_qualified_name(normalize::unresolved_object_name(name.to_owned())?)?
     };
+
+    // check for an object in the catalog with this same name
+    let partial_name = PartialObjectName::from(scx.catalog.resolve_full_name(&name));
+    if let (false, Ok(item)) = (if_not_exists, scx.catalog.resolve_item(&partial_name)) {
+        return Err(PlanError::ItemAlreadyExists {
+            name: name.item,
```

Comment:
I think it would be better if we used the full name for all of these error messages. That way the error has a fully qualified name.